File: Online_Judge/100 - The 3n + 1 problem/main.cpp
#include <iostream>

int main() {
    int x, y = 0;
    while (std::cin >> x >> y) {
        int max = 0;
        int min = 0;
        if (x > y) {
            max = x;
            min = y;
        } else {
            max = y;
            min = x;
        }
        int maxCycle = 0;
        for (int i = min; i <= max; i++) {
            int cycle = 1;
            int n = i;
            while (n != 1) {
                if (n % 2 == 0) {
                    n /= 2;
                } else {
                    n = 3 * n + 1;
                }
                cycle++;
            }
            if (cycle > maxCycle) {
                maxCycle = cycle;
            }
        }
        std::cout << x << " " << y << " " << maxCycle << std::endl;
    }
}

File: Online_Judge/100 - The 3n + 1 problem/main.py
def nb_cycle(n):
    r = 1
    while n != 1:
        if n%2==0:
            n//=2
        else:
            n=n*3+1
        r+=1
    return r

for i in open(0):
    x, y = map(int, i.split())
    i,j = min(x,y),max(x,y)
    print(x,y,max(nb_cycle(a) for a in range(i,j+1)))
